tests/viz/test_edge_connections.py

```
@@ -0,0 +1,717 @@
+
+        node_ids = set(nodes.keys())
+        # Should have inner (collapsed) and validate visible
+        # Hierarchical IDs: middle/inner and middle/validate
+        assert "middle/inner" in node_ids or "middle/validate" in node_ids, (
+            f"Expected middle/inner or middle/validate in nodes: {node_ids}"
+        )
+
+    def test_outer_depth1_input_routes_to_inner(self, page):
+        """Internal-only inputs are hidden when inner is collapsed."""
+        graph = make_outer()
+        nodes, edges = extract_geometries(page, graph, depth=1)
```

Ruff (0.14.14)

[warning] Unpacked variable `nodes` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)